client/src/components/Ratings/Review/ReviewTile.jsx

```diff
@@ -1,27 +1,42 @@
-import React, { useState } from 'react';
+/* eslint-disable react/prop-types */
+// TODO: fix pro-types for this file
+import React, { useState, useEffect } from 'react';
 import moment from 'moment';
+import PropTypes from 'prop-types';
 import styled from 'styled-components';
 import Star from '../../../Star';
 import PhotoList from './PhotoList';
+import ReviewContent from './ReviewContent';
+import Helpfulness from './Helpfulness';
 
-export default function ReviewTile(review) {
+export default function ReviewTile({ review, addHelpVote, reportReview, index }) {
   // TODO: display a photo; sort, response, recp,,emt
   const {
-    review: {
-      rating, summary, recommend, response, date, body, photos,
-    },
+    rating, summary, recommend, response, date, body, photos, helpfulness,
   } = review;
   const convertedDate = moment(date).format('MMMM D, YYYY');
   // eslint-disable-next-line react/destructuring-assignment
-  const usernameDate = `${review.review.reviewer_name},  ${convertedDate}`;
+  const reviewId = review.review_id;
+  // eslint-disable-next-line react/destructuring-assignment
+  const usernameDate = `${review.reviewer_name},  ${convertedDate}`;
   const [showModal, setShowModal] = useState(false);
   const [modalUrl, setModalUrl] = useState('');
 
-  const clickPhoto = (index) => {
-    // console.log('clicked');
-    setModalUrl(photos[index].url);
+  const clickPhoto = (i) => {
+    setModalUrl(photos[i].url);
     setShowModal(!showModal);
   };
+
+  useEffect(() => {
+    const close = (e) => {
+      if (e.key === 'Escape') {
```

Comment:
Wondering if you might want to make it so when they click outside of the modal it closes too. You can use `e.currentTarget` instead of `e.target` to make sure they aren't clicking on any of the children elements of the outside wrapper

Reply:
I made it half-work, now it's click everywhere exit, will take another look later.